lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -593,9 +604,9 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
             # since when an IP change happens, "_on_peer_relation_joined" won't be called,
             # we need to alert the leader that it must recompute the node roles for any unit whose
             # roles were changed while the current unit was cut-off from the rest of the network
-            self.on[PeerRelationName].relation_joined.emit(
-                self.model.get_relation(PeerRelationName)
-            )
+            self._on_peer_relation_joined(event)
+            if event.deferred:
+                return
```

Comment:
i'm not sure here, the rest of the hook may be not-invoked when very much needed to be (i.e plugin_manager.run)

Reply:
Maybe the `if` can indeed go away. The `emit()` is equal to calling the `self._on_peer_relation_joined`

Reply:
yes, my concern is more about the `return`